src/isomorphic/classic/types/ReactPropTypes.js

```diff
@@ -226,11 +227,15 @@ function createInstanceTypeChecker(expectedClass) {
 
 function createEnumTypeChecker(expectedValues) {
   if (!Array.isArray(expectedValues)) {
-    return createChainableTypeChecker(function() {
-      return new Error(
-        `Invalid argument supplied to oneOf, expected an instance of array.`
-      );
-    });
+    if (__DEV__) {
+      warning(false, 'Invalid argument supplied to oneOf, expected an instance of array.');
```

Comment:
We would still want to return a function for consistency—it would just be a no-op.